df_py/predictoor/models.py

```diff
@@ -71,14 +70,72 @@ def accuracy(self):
         return self._accuracy
 
 
+class PredictionSummary:
+    @enforce_types
+    def __init__(self, prediction_count, correct_prediction_count, contract_addr):
+        self.prediction_count = prediction_count
+        self.correct_prediction_count = correct_prediction_count
+        self.contract_addr = contract_addr
+
+    @property
+    def accuracy(self) -> float:
+        if self.prediction_count == 0:
+            return 0
+        return self.correct_prediction_count / self.prediction_count
+
+
 class Predictoor(PredictoorBase):
     @enforce_types
     def __init__(self, address: str):
         super().__init__(address, 0, 0, 0)
         self._predictions: List[Prediction] = []
 
+    def get_prediction_summary(self, contract_addr: str) -> PredictionSummary:
+        """
+        Get the prediction summary for a specific contract address.
+
+        @param contract_addr: The contract address for which to get the prediction summary.
+
+        @return:
+            PredictionSummary - The prediction summary for the specified contract address.
+        """
+        prediction_count = 0
+        correct_prediction_count = 0
+        for prediction in self._predictions:
+            if prediction.contract_addr == contract_addr:
+                prediction_count += 1
+                if prediction.is_correct:
+                    correct_prediction_count += 1
+
+        return PredictionSummary(
+            prediction_count, correct_prediction_count, contract_addr
+        )
+
+    @property
+    def prediction_summaries(self) -> Dict[str, PredictionSummary]:
+        """
+        Get the summaries of all predictions made by this Predictoor.
+
+        @return
+            Dict[str, PredictionSummary] - A dict of PredictionSummary objects.
+        """
+        unique_contract_addresses = set(
```

Comment:
I would simplify this and just:
```
prediction_summaries = {}
for prediction in self._predictions:
    contract_addr = prediction.contract_addr
    if contract_addr in prediction_summaries:
       continue
    prediction_summaries[contract_addr] = self.get_prediction_summary(contract_addr)
    
return prediction_summaries
```
to avoid double iteration